test-clients/src/main/java/com/hedera/services/bdd/suites/contract/OCTokenSpec.java

```diff
@@ -0,0 +1,270 @@
+	private static final String APPROVE_ABI = "{\"constant\":false,\"inputs\":[{\"name\":\"_spender\",\"type\":\"address\"},{\"name\":\"_value\",\"type\":\"uint256\"}],\"name\":\"approve\",\"outputs\":[{\"name\":\"success\",\"type\":\"bool\"}],\"payable\":false,\"stateMutability\":\"nonpayable\",\"type\":\"function\"}";
+	private static final String TRANSFER_FROM_ABI = "{\"constant\":false,\"inputs\":[{\"name\":\"_from\",\"type\":\"address\"},{\"name\":\"_to\",\"type\":\"address\"},{\"name\":\"_value\",\"type\":\"uint256\"}],\"name\":\"transferFrom\",\"outputs\":[{\"name\":\"success\",\"type\":\"bool\"}],\"payable\":false,\"stateMutability\":\"nonpayable\",\"type\":\"function\"}";
+	private static final String SYMBOL_ABI =   "{\"constant\":true,\"inputs\":[],\"name\":\"symbol\",\"outputs\":[{\"name\":\"\",\"type\":\"string\"}],\"payable\":false,\"stateMutability\":\"view\",\"type\":\"function\"}";
+	private static final String DECIMALS_ABI = "{\"constant\":true,\"inputs\":[],\"name\":\"decimals\",\"outputs\":[{\"name\":\"\",\"type\":\"uint8\"}] ,\"payable\":false,\"stateMutability\":\"view\",\"type\":\"function\"}";
+
+	public static void main(String... args) {
+		new org.ethereum.crypto.HashUtil();
+
+		new OCTokenSpec().runSuiteSync();
+	}
+
+	@Override
+	protected List<HapiApiSpec> getSpecsInSuite() {
+		return List.of(new HapiApiSpec[] {
+				ocToken(),
+		});
+	}
+
+	private long getLongValueFromRegistry(HapiApiSpec spec, String from, CallTransaction.Function function) {
+		byte[] 	balanceRawBytes = spec.registry().getBytes(from);
+
+		long balance = 0;
+		if(balanceRawBytes != null && balanceRawBytes.length > 0) {
+			Object[] retResults = function.decodeResult(balanceRawBytes);
+			if (retResults != null && retResults.length > 0) {
+				BigInteger retBi = (BigInteger) retResults[0];
+				balance = retBi.longValue();
+			}
+		}
+		return balance;
+	}
+
+	HapiApiSpec ocToken() {
+
+		return defaultHapiSpec("ocToken")
+				.given(
+						cryptoCreate("tokenIssuer").balance( 10_000_000_000_000L),
+						cryptoCreate("Alice").balance( 10_000_000_000L).payingWith("tokenIssuer"),
+						cryptoCreate("Bob").balance( 10_000_000_000L).payingWith("tokenIssuer"),
+						cryptoCreate("Carol").balance( 10_000_000_000L).payingWith("tokenIssuer"),
+						cryptoCreate("Dave").balance( 10_000_000_000L).payingWith("tokenIssuer"),
+
+						getAccountInfo("tokenIssuer").saveToRegistry("tokenIssuerAcctInfo"),
+						getAccountInfo("Alice").saveToRegistry("AliceAcctInfo"),
+						getAccountInfo("Bob").saveToRegistry("BobAcctInfo"),
+						getAccountInfo("Carol").saveToRegistry("CarolAcctInfo"),
+						getAccountInfo("Dave").saveToRegistry("DaveAcctInfo"),
+
+						fileCreate("bytecode")
+								.path(PATH_TO_OC_TOKEN_BYTECODE),
+
+						contractCreate("tokenContract", TOKEN_ERC20_CONSTRUCTOR_ABI,
+								1_000_000L, "OpenCrowd Token", "OCT")
+								.gas(250_000L)
```

Comment:
How did you get through the `GAS_LIMIT_EXCEEDED` here ?

Reply:
The root cause is in the HapiContractCreate class. The encoding needs to be encodeArguments(...) instead of encode(...).